pkg/dataset/config.go

```diff
@@ -0,0 +1,48 @@
+package dataset
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgmodel/common/schema"
+	"gopkg.in/yaml.v2"
+)
+
+const defaultChunkInterval = 8 * time.Hour
+
+var setDefaultChunkIntervalSQL = fmt.Sprintf("SELECT %s.set_default_chunk_interval($1)", schema.Prom)
+
+type Config struct {
+	Metrics Metrics `yaml:"metrics"`
+}
+
+type Metrics struct {
+	ChunkInterval time.Duration `yaml:"default_chunk_interval"`
+}
+
+func NewConfig(file string) (Config, error) {
+	cfg := Config{}
+
+	data, err := ioutil.ReadFile(file) // #nosec G304
+	if err != nil {
+		return cfg, err
+	}
+
+	err = yaml.Unmarshal(data, &cfg)
+	return cfg, err
+}
+
+func (c *Config) Apply(conn *pgx.Conn) error {
+	if c.Metrics.ChunkInterval == 0 {
```

Comment:
```suggestion
	if c.Metrics.ChunkInterval <= 0 {
```

Let's correct for potential negative values.